src/txn/txn_recover.c

```diff
@@ -531,14 +532,16 @@ __wt_txn_recover(WT_SESSION_IMPL *session)
     config = NULL;
     do_checkpoint = hs_exists = true;
     eviction_started = false;
+    oldest_ts = WT_TS_NONE;
     was_backup = F_ISSET(conn, WT_CONN_WAS_BACKUP);
 
     /* We need a real session for recovery. */
     WT_RET(__wt_open_internal_session(conn, "txn-recover", false, WT_SESSION_NO_LOGGING, &session));
     r.session = session;
     WT_MAX_LSN(&r.max_ckpt_lsn);
     WT_MAX_LSN(&r.max_rec_lsn);
-    conn->txn_global.recovery_timestamp = conn->txn_global.meta_ckpt_timestamp = WT_TS_NONE;
+    conn->txn_global.recovery_timestamp = conn->txn_global.meta_ckpt_timestamp =
+      conn->txn_global.oldest_ckpt_hs_timestamp = WT_TS_NONE;
```

Comment:
we may not need this explicit assignment to NONE as oldest_ts is already initialized, so changing it as follows can fix.
```
    if (hs_exists)
        WT_ERR_NOTFOUND_OK(
          __wt_meta_get_oldest_ckpt_timestamp(session, WT_HS_URI, &oldest_ts), true);
    conn->txn_global.oldest_ckpt_hs_timestamp = oldest_ts;
```

Reply:
My concern was that the function could potentially exit before `conn->txn_global.oldest_ckpt_hs_timestamp` is set, leaving the value uninitialised. But I'm happy to remove if we are sure the value won't be used in this scenario.

Reply:
Ok. Understood.